### ISSUE TITLE
Resizing of PNG24 images - transparency?

### ISSUE THREAD

**Issue by flobox** [mosbth/cimage#117]
I am havin some troubles with transparent PNG24 images.

Attached you find the original PNG24.
I resized the original PNG using cimage and a different image resizer with used before ( imageresizing.net ).

Unfortunately it seems that cimage is producing a PNG8 version?! The transparency isn't as good as within the original file - as you can see.

Any hints?!
![original](https://cloud.githubusercontent.com/assets/3763232/10580230/cd82b886-767c-11e5-8af9-24ec9fd16522.png)
![imageresizer](https://cloud.githubusercontent.com/assets/3763232/10580228/cd7bc242-767c-11e5-875e-a3742683bdf6.png)
![cimage](https://cloud.githubusercontent.com/assets/3763232/10580229/cd7c9424-767c-11e5-9c5d-d55e375c5d14.png)

**Comment by flobox** [comment]
The first is the original. The second the imageresizing.net version. The third the cimage version.

I used the following operators: ?w=190&h=65

**Comment by mosbth** [comment]
Thanks for the test images, I'll check it out.

**Comment by mosbth** [comment]
Ok, this far I come for now. The first thing I did was to prevent the image from being handled as a palette-image (change in CImage for png-images with few colors).

Then I added a test option `&no-resample` for setting the default sample strategy on resize, (in branch 077), saying "use imagecopyresize instead of imagecopyresampled".

Resampled (as default):
`img.php?src=issue117/tri_original.png&w=190&nc`
![resample](https://cloud.githubusercontent.com/assets/169550/10606265/03e583ee-7732-11e5-8172-0ce95d134c9a.png)

Above makes text look edgy/blurry.

Resize without resample:
`img.php?src=issue117/tri_original.png&w=190&no-resample&nc`
![resize](https://cloud.githubusercontent.com/assets/169550/10606263/fe2b4b28-7731-11e5-8c55-7c1d1a5a8fac.png)

Above looks more or less like the imageresizing.net version and a version from ImageMagick I used as additional test.

Seems like the result is better using [imagecopyresize](http://php.net/manual/en/function.imagecopyresized.php) instead of [imagecopyresampled](http://php.net/manual/en/function.imagecopyresampled.php), on this particular test.

So, it can look good, but we further need to investigate how to "always" make it look good.

The compare-tool is useful for investigating this kind of stuff, a testprogram for this issue is in `webroot/compare/issue117-PNG24.php`.
